src/stream/src/executor/exchange/permit.rs

```diff
@@ -165,6 +215,10 @@ impl Receiver {
         if let Some(permits) = permits {
             self.permits.add_permits(permits);
         }
+        if let Message::Chunk(chunk) = &message {
+            self.metric_memory_size.sub(chunk.estimated_size() as i64);
+            self.metric_num_rows.sub(chunk.cardinality() as i64);
+        }
```

Comment:
Interesting. Does this also work for remote exchange? What will happen if we operate on the same metric with the same labels in different nodes?

Reply:
> Does this also work for remote exchange?

I think so. But for remote exchange, these metrics reflect the total size **inside the channel**, not only the memory usage, because some chunks may be inflight on the network.

> What will happen if we operate on the same metric with the same labels in different nodes?

I guess they can be accumulated properly since we only use `add` and `sub`.